File: app/src/main/java/com/gs/buluo/store/presenter/GoodsPresenter.java
package com.gs.buluo.store.presenter;

import com.gs.buluo.common.network.BaseSubscriber;
import com.gs.buluo.store.bean.GoodList;
import com.gs.buluo.store.bean.ResponseBody.BaseResponse;
import com.gs.buluo.store.network.GoodsApis;
import com.gs.buluo.store.network.TribeRetrofit;
import com.gs.buluo.store.view.impl.IGoodsView;

import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by hjn on 2016/11/16.
 */
public class GoodsPresenter extends BasePresenter<IGoodsView> {
    private String nextSkip;

    public void getGoodsList() {
        TribeRetrofit.getInstance().createApi(GoodsApis.class).getGoodsListFirst(20)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new BaseSubscriber<BaseResponse<GoodList>>() {
                    @Override
                    public void onNext(BaseResponse<GoodList> goodListBaseResponse) {
                        super.onNext(goodListBaseResponse);
                        nextSkip = goodListBaseResponse.data.nextSkip;
                        if (isAttach()) mView.getGoodsInfo(goodListBaseResponse.data);
                    }
                });
    }

    public void loadMore() {
        TribeRetrofit.getInstance().createApi(GoodsApis.class).getGoodsList(20,nextSkip)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new BaseSubscriber<BaseResponse<GoodList>>() {
                    @Override
                    public void onNext(BaseResponse<GoodList> goodListBaseResponse) {
                        super.onNext(goodListBaseResponse);
                        if (isAttach()) mView.getGoodsInfo(goodListBaseResponse.data);
                    }
                });
    }
}

File: app/src/main/java/com/gs/buluo/store/kotlin/presenter/GoodsPresenter.kt
package com.gs.buluo.store.kotlin.presenter

import com.gs.buluo.common.network.BaseResponse
import com.gs.buluo.store.bean.GoodList

/**
 * Created by hjn on 2016/11/16.
 */
class GoodsPresenter(var mView: com.gs.buluo.store.view.impl.IGoodsView) : KotBasePresenter() {
    private var nextSkip: String? = null

    fun getGoodsList() {
        com.gs.buluo.store.network.TribeRetrofit.getInstance().createApi(com.gs.buluo.store.network.GoodsApis::class.java).getGoodsListFirst(20)
                .subscribeOn(rx.schedulers.Schedulers.io())
                .observeOn(rx.android.schedulers.AndroidSchedulers.mainThread())
                .subscribe(object : com.gs.buluo.common.network.BaseSubscriber<BaseResponse<GoodList>>() {
                    override fun onNext(goodListBaseResponse: com.gs.buluo.common.network.BaseResponse<GoodList>?) {
                        super.onNext(goodListBaseResponse)
                        nextSkip = goodListBaseResponse!!.data.nextSkip
                        mView.getGoodsInfo(goodListBaseResponse.data)
                    }
                })
    }

    fun loadMore() {
        com.gs.buluo.store.network.TribeRetrofit.getInstance().createApi(com.gs.buluo.store.network.GoodsApis::class.java).getGoodsList(20, nextSkip)
                .subscribeOn(rx.schedulers.Schedulers.io())
                .observeOn(rx.android.schedulers.AndroidSchedulers.mainThread())
                .subscribe(object : com.gs.buluo.common.network.BaseSubscriber<BaseResponse<GoodList>>() {
                    override fun onNext(goodListBaseResponse: com.gs.buluo.common.network.BaseResponse<GoodList>?) {
                        super.onNext(goodListBaseResponse)
                        mView.getGoodsInfo(goodListBaseResponse!!.data)
                    }
                })
    }
}
